ofrak_core/ofrak/core/free_space.py

```diff
@@ -370,15 +358,50 @@ async def _get_partially_overlapping_resources(
         )
 
 
+async def _find_and_delete_overlapping_children(resource: Resource, freed_range: Range):
+    # Note this filter calculation has the potential to be very expensive if, for instance,
+    # the resource is an entire program segment...
+    overlap_resources = list(
+        await resource.get_children_as_view(
+            MemoryRegion,
+            r_filter=ResourceFilter(
+                tags=(MemoryRegion,),
+                attribute_filters=(
+                    ResourceAttributeRangeFilter(MemoryRegion.VirtualAddress, max=freed_range.end),
+                ),
+            ),
+        )
+    )
+    for possible_overlapping_child in overlap_resources:
+        if (
+            possible_overlapping_child.virtual_address > freed_range.start
+            or possible_overlapping_child.end_vaddr() > freed_range.start
+        ):
+            await possible_overlapping_child.resource.delete()
```

Comment:
```suggestion
                attribute_filters=(
                    ResourceAttributeRangeFilter(MemoryRegion.VirtualAddress, max=freed_range.end),
                    ResourceAttributeRangeFilter(MemoryRegion.EndVaddr, min=freed_range.start),
                ),
            ),
        )
    )
    for overlapping_child in overlap_resources:
        await possible_overlapping_child.resource.delete()
```

Indexable attributes have improved a bit since the original implementation. The overlapping children can now be found with one query.

Reply:
Testing the suggested change in my application of this modifier, will take it if results match.